app/controllers/ClaimPeriodStartController.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions._
+import forms.ClaimPeriodStartFormProvider
+import javax.inject.Inject
+import models.{Mode, NormalMode, UserAnswers}
+import navigation.Navigator
+import pages.ClaimPeriodStartPage
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import views.html.ClaimPeriodStartView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ClaimPeriodStartController @Inject()(
+                                        override val messagesApi: MessagesApi,
+                                        sessionRepository: SessionRepository,
+                                        navigator: Navigator,
+                                        identify: IdentifierAction,
+                                        getData: DataRetrievalAction,
+                                        requireData: DataRequiredAction,
+                                        formProvider: ClaimPeriodStartFormProvider,
+                                        val controllerComponents: MessagesControllerComponents,
+                                        view: ClaimPeriodStartView
+                                      )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  private def form = formProvider()
+
+  def onPageLoad: Action[AnyContent] = (identify andThen getData) {
```

Comment:
why do you need `identify` when it's an unauthenticated journey that needs to be built?

Reply:
identify  is like session based identifying - checks whether there is a session or not , the logic is inside `SessionIdentifierAction`

Reply:
it's a strange name for what it is doing then

Reply:
cannot it not be called `getSessionData` or something?

Reply:
can we write something like `(checkSession andThen getData)`?

Reply:
renamed it and removed other logic